apps/authentication/views/login.py

```diff
@@ -227,6 +227,10 @@ def redirect_third_party_auth_if_need(self, request):
         return redirect_url
 
     def get(self, request, *args, **kwargs):
+        next_page = request.GET.get(self.redirect_field_name)
+        if next_page:
+            request.session[self.redirect_field_name] = next_page
+
         if request.user.is_staff:
             first_login_url = redirect_user_first_login_or_index(
                 request, self.redirect_field_name
```

Comment:
The code doesn't contain errors or irregularities in terms of functionality but may benefit from some minor improvements for clarity and possibly to prevent SQL injection attacks.

To improve on readability and avoid SQL injections potentially introduced due to user input, it would be wise to use a safer method like parameterized queries when querying with Django's ORM:

```python
# Avoids SQL Injection through using safe=True
query = (
    "SELECT "
    f"{self.redirect_field_name} FROM auth_session "
    f"WHERE ("
    + ", ".join(
        str(user) for user in request.GET.keys() if "redirect_field_name" not in dir(user)  # Exclude query parameters that have special names
        )
    ") AND is_active="
    True
).
params={

However, this changes the behavior slightly since now only non-query params pass into the database, which might not align perfectly with the initial intention. To mitigate such concerns, you can always add additional checks at other points of data processing based on what kind of information should go directly into the session vs being queried later:

1. At each point where data might need modification (e.g., before redirects), check whether the object has a `next_page` property instead:

```
def get(self, request, *args, **kwargs):    
   ...
   try:
       next_page = getattr(request.user, self.redirect_field_name)
   except AttributeError:
       # No redirect_next provided; use default index path 
...
If these checks are run regularly enough they will greatly reduce the risk of unintentional SQLInjection because we'll only insert data relevant to the current state.
```